dask/tests/test_distributed.py

```diff
@@ -398,3 +335,93 @@ async def test_annotation_pack_unpack(c, s, a, b):
     unpacked_hlg = HighLevelGraph.__dask_distributed_unpack__(packed_hlg, annotations)
     annotations = unpacked_hlg["annotations"]
     assert annotations == {"workers": {"n": ("alice",)}}
+
+
+class SchedulerImportCheck(SchedulerPlugin):
+    """Plugin to help record which modules are imported on the scheduler"""
+
+    def __init__(self, pattern):
+        self.pattern = pattern
+
+    async def start(self, scheduler):
+        # Record the modules that have been imported when the scheduler starts
+        self.start_modules = set()
+        for mod in set(sys.modules):
+            if not mod.startswith(self.pattern):
+                self.start_modules.add(mod)
+            else:
+                # Maually remove the target library
+                sys.modules.pop(mod)
```

Comment:
Modifying `sys.modules` in this way makes me a little uncomfortable -- though maybe this is actually fine

Since the `cluster` fixture creates a new process with the launched scheduler I wouldn't expect anything that's not a `distributed` dependency to be imported. What happens if we instead just `assert` that none of the initial `sys.modules` start with `self.pattern`?

Reply:
Sorry - I should have commented about this earlier.  I did start off by adding the assertion you suggest above.  However, the test was failing for the `_array_creation` case (pandas is not a problem).  It seems that numpy is imported somewhere outside of this test, and so I added the logic to manually "clean" `sys.modules` (rather than looking into it further) :/